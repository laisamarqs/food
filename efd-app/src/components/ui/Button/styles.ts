import styled from 'styled-components'
import { ButtonProps } from '.'
import { colors } from '../../../styles'

export const ButtonContainer = styled.button<Omit<ButtonProps, 'label'>>`
  color: ${(props) =>
    props.background === 'primary' ? colors.primaryLight : colors.primaryDark};
  background-color: ${(props) =>
    props.background === 'primary' ? colors.primaryDark : colors.primaryLight};
  border: none;
  text-align: center;
  font-size: 0.875rem;
  font-weight: 700;
  cursor: pointer;
  padding: 0.5rem 0.5rem;

  &:hover {
    opacity: 0.8;
  }
`