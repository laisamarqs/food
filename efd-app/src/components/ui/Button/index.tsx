import { ButtonContainer } from './styles'

export type ButtonProps = {
  type: 'button' | 'submit' | 'reset'
  background: 'primary' | 'secundary'
  label: string
  title: string
  onClick?: React.MouseEventHandler<HTMLButtonElement>
}

const Button = ({ type, background, label, title, onClick }: ButtonProps) => (
  <ButtonContainer
    background={background}
    title={title}
    type={type}
    onClick={onClick}
  >
    {label}
  </ButtonContainer>
)

export default Button