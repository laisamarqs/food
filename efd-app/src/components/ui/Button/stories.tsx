import type { Meta, StoryObj } from '@storybook/react'

import Button from '.'

const meta = {
  title: 'ui/Button',
  component: Button,
  parameters: {
    layout: 'centered'
  },
  argTypes: {
    onClick: { control: { disable: true } }
  },
  tags: ['autodocs']
} satisfies Meta<typeof Button>

// eslint-disable-next-line react-refresh/only-export-components
export default meta
type Story = StoryObj<typeof Button>

export const Primary: Story = {
  args: {
    label: 'Button primary',
    background: 'primary',
    title: 'Click me',
    type: 'button'
  }
}

export const Secondary: Story = {
  args: {
    label: 'Button secundary',
    background: 'secundary',
    title: 'Click me',
    type: 'button'
  }
}