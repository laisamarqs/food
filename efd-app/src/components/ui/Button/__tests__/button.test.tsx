import '@testing-library/jest-dom'
import { expect, vi } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'

import Button from '..'

describe('<Button />', () => {
  it('should render a primary button"', () => {
    const { container } = render(
      <Button
        label="button primary"
        background="primary"
        type="button"
        title="click me"
      />
    )
    const buttonPrimary = screen.getByRole('button', {
      name: /button primary/i
    })
    expect(buttonPrimary).toHaveStyle('color:  rgb(255, 235, 217)')
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a secondary button"', () => {
    const { container } = render(
      <Button
        label="button secondary"
        background="secundary"
        type="button"
        title="click me"
      />
    )
    const buttonSecondary = screen.getByRole('button', {
      name: /button secondary/i
    })
    expect(buttonSecondary).toHaveStyle('color:   rgb(177, 57, 57)')
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should call function on button click', () => {
    const handleClick = vi.fn()
    render(
      <Button
        label="button"
        background="secundary"
        type="button"
        title="click me"
        onClick={handleClick}
      />
    )
    const button = screen.getByRole('button', {
      name: /button/i
    })
    fireEvent.click(button)

    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})