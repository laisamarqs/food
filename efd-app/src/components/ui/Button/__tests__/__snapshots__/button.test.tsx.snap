// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should render a primary button" 1`] = `
<button
  class="sc-dcJsrY bgvNbD"
  title="click me"
  type="button"
>
  button primary
</button>
`;

exports[`<Button /> > should render a secondary button" 1`] = `
<button
  class="sc-dcJsrY bDWePn"
  title="click me"
  type="button"
>
  button secondary
</button>
`;